docs/configuration.rst

```diff
@@ -0,0 +1,233 @@
+Configuration options
+#####################
+
+The Raven client can be configured through a bunch of configuration options.
+Some of them are automatically populated by environment variables when a new
+instance of the ``Configuration`` class is created. The following is an exhaustive
+list of them with a description of what they do and some example values.
+
+Send attempts
+=============
+
+The number of attempts that should should be made to send an event before erroring
+and dropping it from the queue.
+
+.. code-block:: php
+
+    $configuration = new Configuration(['send_attempts' => 1]);
+    $configuration->setSendAttempts(1);
+
+By default this option is set to 6.
+
+Server
+======
+
+The DSN of the Sentry server the authenticated user is bound to.
+
+.. code-block:: php
+
+    // Before Sentry 1.9 the private DSN must be used
+    $configuration = new Configuration(['server' => 'http://public:private@example.com/1']);
+
+    // After Sentry 1.9 the public DSN must be used
+    $configuration = new Configuration(['server' => 'http://public@example.com/1']);
+
+By default this option is borrowed from the ``SENTRY_DSN`` environment variable,
+if it exists. A value of ``null`` disables completely the sending of any event.
+Since Sentry 1.9 the public DSN must be used instead of the private one, which
+is deprecated and whose support will be removed in future releases of the server.
```

Comment:
I really think there should be at least an alias for the `dsn` key, although I think `server` should be the alias (or not used at all).

DSN is the common language to refer to this information for every integration AFAIK.

Reply:
I strongly agree with you, however the change is out of scope of this PR which is only for documentation. I will open a new one to address the comment asap